src/Bot/Services/LogAnalysis/RyuLogReader.cs

```diff
@@ -0,0 +1,256 @@
+        List<string> errors = new List<string>();
+        List<string> currentErrorsLines = new List<string>();
+        bool errorLine = false;
+
+        foreach (string line in log.RawLogContent.Split("\n"))
+        {
+            if (line.IsNullOrWhitespace())
+            {
+                continue;
+            }
+
+            if (line.Contains("|E|"))
+            {
+                currentErrorsLines = [line];
+                errors.AddRange(currentErrorsLines);
+                errorLine = true;
+            }
+            else if (errorLine && line[0].ToString() == "")
+            {
+                currentErrorsLines.AddRange(line);
+            }
+        }
+
+        if (currentErrorsLines.Count > 0)
+        {
+            errors.AddRange(currentErrorsLines);
+        }
+        
+        log.LogErrors = errors;
+    }
+
+    string[] sizes = ["KB", "KiB", "MB", "MiB", "GB", "GiB"];
+
+    void GetHardwareInfo()
+    {
+        // CPU
+        Match cpuMatch = Regex.Match(log.RawLogContent, @"CPU:\s([^;\n\r]*)", RegexOptions.Multiline);
+        if (cpuMatch.Success)
+        {
+            log.Hardware.Cpu = cpuMatch.Groups[1].Value.TrimEnd();
+        }
+        else
+        {
+            log.Hardware.Cpu = "Unknown";
+        }
+        
+        // RAM
+        Match ramMatch = Regex.Match(log.RawLogContent,
+            @$"RAM: Total ([\d.]+) ({sizes}) ; Available ([\d.]+) ({sizes})", RegexOptions.Multiline);
+        if (ramMatch.Success)
+        {
+            double ramAvailable = ConvertGiBtoMiB(Convert.ToDouble(ramMatch.Groups[3].Value));
+            double ramTotal = ConvertGiBtoMiB(Convert.ToDouble(ramMatch.Groups[1].Value));
+            log.Hardware.Ram = $"{ramAvailable:.0f}/{ramTotal:.0f} MiB";
+        }
+        else
+        {
+            log.Hardware.Ram = "Unknown";
+        }
+
+        // Operating System (OS)
+        Match osMatch = Regex.Match(log.RawLogContent, @"Operating System:\s([^;\n\r]*)", RegexOptions.Multiline);
+        if (osMatch.Success)
+        {
+            log.Hardware.Os = osMatch.Groups[1].Value.TrimEnd();
+        }
+        else
+        {
+            log.Hardware.Os = "Unknown";
+        }
+
+        // GPU
+        Match gpuMatch = Regex.Match(log.RawLogContent, @"PrintGpuInformation:\s([^;\n\r]*)",
+            RegexOptions.Multiline);
+        if (gpuMatch.Success)
+        {
+            log.Hardware.Gpu = gpuMatch.Groups[1].Value.TrimEnd();
+            // If android logs starts showing up, we can detect android GPUs here
+        }
+        else
+        {
+            log.Hardware.Gpu = "Unknown";
+        }
+    }
+
+    void GetEmuInfo()
+    {
+        // Ryujinx Version check
+        foreach (string line in log.RawLogContent.Split("\n"))
+        {
+            // Greem's Stable build
+            if (LogAnalysisPatterns.StableVersion.IsMatch(line))
+            {
+                log.EmulatorInfo.Version = (RyujinxVersion.Stable, line.Split("\n")[-1].Trim());
```

Comment:
I don't think this will work; because the content is already being split on each newline, no newlines can be present in any of the strings by Split()